chia/util/condition_tools.py

```diff
@@ -65,6 +66,28 @@ def conditions_by_opcode(
     return d
 
 
+def pkm_pairs(npc_list: List[NPC], additional_data: bytes) -> Tuple[List[G1Element], List[bytes]]:
+    ret: Tuple[List[G1Element], List[bytes]] = ([], [])
+
+    for npc in npc_list:
+        for opcode, l in npc.conditions:
+            if opcode == ConditionOpcode.AGG_SIG_UNSAFE:
+                for cwa in l:
+                    assert len(cwa.vars) == 2
+                    assert len(cwa.vars[0]) == 48 and len(cwa.vars[1]) <= 1024
+                    assert cwa.vars[0] is not None and cwa.vars[1] is not None
+                    ret[0].append(G1Element.from_bytes(cwa.vars[0]))
+                    ret[1].append(cwa.vars[1])
+            elif opcode == ConditionOpcode.AGG_SIG_ME:
+                for cwa in l:
+                    assert len(cwa.vars) == 2
+                    assert len(cwa.vars[0]) == 48 and len(cwa.vars[1]) <= 1024
+                    assert cwa.vars[0] is not None and cwa.vars[1] is not None
+                    ret[0].append(G1Element.from_bytes(cwa.vars[0]))
+                    ret[1].append(cwa.vars[1] + npc.coin_name + additional_data)
+    return ret
```

Comment:
this is the main part of this patch. This interface simplifies call sites right now, but it also lays the foundation for a simpler implementation once the `NPCResult` data structure is simplified

Reply:
that's correct. Although, the main reason they're here is because we have them in the `pkm_pairs_for_conditions_dict()` version of this function. Here they are "true" assertions though, things that should never happen.